### DISCUSSION TITLE
Get pipewire filter chain from easyeffects

### DISCUSSION THREAD

**Discussion by ghost** [wwmm/easyeffects#2709] [Q&A] [answered]
Is it possible to get the pipewire filter chain that Easyeffects creates and then run them manually to use with pipewire directly instead of using Easyeffects? This could allow for usage of filters when Easyeffects cannot be opened, I.E when running on a tty.
This is nice, as Easyeffects provides a intuitive GUI to configure a filter chain, while writing them manually is quite complicated.

**Comment by wwmm** [comment, marked as answer]
EasyEffects does not use PipeWire's filter chain. What we do is using `pw_filter` objects to write or wrap filters and create our pipeline. Just like an audio player stream only exists while the player is running our chain of `pw_filter` objects only exists while EasyEffects is running.

I have never looked at PipeWire's filter chain in detail but it does not feel like something that is suitable for EasyEffects usage.

**Reply by ghost** [reply]
Ahh, I guess I misunderstood the README, this makes my question invalid, thanks.
